### PR TITLE
fs/nvs: Improve init speed and remove fs->locked

### PR DESCRIPTION
**Pull request by Laczen**
This patch removes the free space calculation from nvs initialization.
The available space can be calculated if required using the routine
nvs_calc_free_space.

This patch also removes the locked state of nvs, it is not possible to
get in a locked state.

Changes:

Removed locked state and free_space from the nvs structure.

nvs_reinit(): has been replaced with by an internal only function
_nvs_startup().

nvs_write(): removed the possibility to place the file system in a
locked state, if to many gc operations are required it will return
-ENOSPC.

ssize_t nvs_calc_free_space(): introduced, calculates the free space
that is available in the nvs file system.

Signed-off-by: Laczen JMS <laczenjms@gmail.com>

### CODE REVIEW

**Review by dbkinder — APPROVED** [review]
doc changes LGTM

**Comment by Laczen** [comment]
@aurel32, @nvlsianpu, if you have some time could you please review.

**Review by nvlsianpu — COMMENTED** [review]
LGTM. Just one small nip-tick regards logger usage.

File: subsys/fs/nvs/nvs.c [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 #define LOG_LEVEL CONFIG_NVS_LOG_LEVEL
 #include <logging/log.h>
-LOG_MODULE_REGISTER(fs_nvs);
+LOG_MODULE_REGISTER(fs_nvs, LOG_LEVEL);
```
[the remark this answers is not in the document]
Comment (Laczen) [reply subsys/fs/nvs/nvs.c]: Changed, thanks for the review

**Comment by nvlsianpu** [comment]
@Laczen Can you resolve conflicts and fix shippable?

**Comment by Laczen** [comment]
@nvlsianpu, how do I resolve the conflicts (they are also causing the shippable error. I think this is a result of the files being changed by someone in the time between the PR and now.

**Comment by nvlsianpu** [comment]
Do rebase on newest master:

Update your master to the zephyrproject master head (perfectly using --ff-only)
Then checkout nvs_init branch
do `git rebase master` - It will cause conflict which you will resolve by hand:
- for each conflict reslove conflict in file and add it to commit scope (don't do the commit!)
- after all conflict are resolved and staged, continue rebase by `git rebase --continue.

Don't worry if you make any foult while resolving - do `git rebase --abor', and start from the beginning.

**Comment by Laczen** [comment]
@nvlsianpu, I guess I did something wrong because there seems to be a Merge branch of 'master'.

**Review by nvlsianpu — APPROVED** [review]

**Pull request merged**